### PR TITLE
Add UrlPicker component

### PR DESCRIPTION
**Pull request by helen**
This PR likely still needs some code styling to match the rest of the components, as of this writing it is currently just fully pulled over from another project with that name replaced.

### Description of the Change
Adding UrlPicker component as used on multiple projects. Code is largely from @dinhtungdu. The UrlPicker is currently a PlainText component with button icons to set the URL via popover and to warn when a URL is missing. The button icon is used rather than a toolbar button in order to disambiguate when there are multiple links within a block.

### Alternate Designs
Takes cues from how button links work, minus the toolbar because no formatting options.

### Benefits
Reusable component particularly handy for lists of links or CTAs.

### Possible Drawbacks
Not sure.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: components/UrlPicker/index.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+   Button,
+   Tooltip,
+ } from '@wordpress/components';
+ import {
+   PlainText,
+   __experimentalLinkControl as LinkControl,
+ } from '@wordpress/block-editor';
+ import { rawShortcut } from '@wordpress/keycodes';
+ import { debounce } from 'lodash';
+ import { searchablePostTypes } from './helper';
```
Comment (fabiankaegy) [inline components/UrlPicker/index.js]: @helen Seems this function is missing here. Would you be able to also add it to this file? I cam take care of the refactor then :)

Comment (helen) [reply components/UrlPicker/index.js]: Added it as an inline const :)

**Comment by fabiankaegy** [comment]
I had a quick chat with @xavortm about this and he is going to take a look at taking the Component Helen provided here and refactoring it to match the style of the rest of the project. 

We have created a new issue for it here #62 and are going to create a new branch off of this PR to integrate it into a new PR.

**Comment by fabiankaegy** [comment]
Closing this PR in favor of #66

**Pull request closed without merging**